components/timezone/src/ids.rs

```diff
@@ -209,27 +211,30 @@ impl TimeZoneIdMapperBorrowed<'_> {
     /// let mapper = TimeZoneIdMapper::new();
     /// let mapper = mapper.as_borrowed();
     ///
-    /// let result = mapper.normalize_iana("Asia/CALCUTTA").unwrap();
+    /// let result = mapper.normalize_iana("Asia/CALCUTTA");
     ///
     /// assert_eq!(result.0, "Asia/Calcutta");
     /// assert!(matches!(result.0, Cow::Owned(_)));
     /// assert_eq!(*result.1, "inccu");
     ///
     /// // Borrows when able:
-    /// let result = mapper.normalize_iana("America/Chicago").unwrap();
+    /// let result = mapper.normalize_iana("America/Chicago");
     /// assert_eq!(result.0, "America/Chicago");
     /// assert!(matches!(result.0, Cow::Borrowed(_)));
     ///
     /// // Unknown IANA time zone ID:
-    /// assert_eq!(mapper.normalize_iana("America/San_Francisco"), None);
+    /// assert_eq!(mapper.normalize_iana("America/San_Francisco").1, TimeZoneBcp47Id::unknown());
     /// ```
-    pub fn normalize_iana<'s>(&self, iana_id: &'s str) -> Option<(Cow<'s, str>, TimeZoneBcp47Id)> {
-        let (trie_value, string) = self.iana_lookup_with_normalization(iana_id, |_| {})?;
-        let Some(bcp47_id) = self.data.bcp47_ids.get(trie_value.index()) else {
-            debug_assert!(false, "index should be in range");
-            return None;
-        };
-        Some((string, bcp47_id))
+    pub fn normalize_iana<'s>(&self, iana_id: &'s str) -> (Cow<'s, str>, TimeZoneBcp47Id) {
+        self.iana_lookup_with_normalization(iana_id, |_| {})
+            .and_then(|(trie_value, string)| {
+                let Some(bcp47_id) = self.data.bcp47_ids.get(trie_value.index()) else {
+                    debug_assert!(false, "index should be in range");
+                    return None;
+                };
+                Some((string, bcp47_id))
+            })
+            .unwrap_or((Cow::Borrowed("Etc/Unknown"), TimeZoneBcp47Id::unknown()))
```

Comment:
Issue: This function should return `None` if the time zone is not found. It should never map it to `Etc/Unknown`. Canonicalization can do that, but not normalization, please.